flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java

```diff
@@ -112,7 +112,8 @@
                             "Defines whether the cluster uses declarative resource management.");
 
     public static boolean isDeclarativeResourceManagementEnabled(Configuration configuration) {
-        return configuration.get(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT)
-                && !System.getProperties().containsKey("flink.tests.disable-declarative");
+        return configuration.contains(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT)
+                        && configuration.get(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT)
+                || !System.getProperties().containsKey("flink.tests.disable-declarative");
```

Comment:
Isn't this logic also a bit flawed? Assume `ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT` has been set to false and `flink.tests.disable-declarative` has not been set. Then this method should return `true` which is a bit weird. I guess it should be something like

```
if (configuration.contains(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT)) {
  return configuration.get(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT);
} else {
  return !System.getProperties().containsKey("flink.tests.disable-declarative");
}
```

Reply:
yes, you are correct; i'll update and rebase the PR.